<a href="https://colab.research.google.com/github/Deunich/Prueba-Z/blob/main/Prueba_Z.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##### -En los negocios, hay tantas decisiones que se deben tomar todos los días, en lugar de tomarlas por corazonada, hacer experimentos y crear hipótesis es un enfoque más centrado en los datos para tomar decisiones en los negocios.


# Prueba de hipótesis

- La hipótesis es un mecanismo para tomar decisiones en estadística inferencial. Ayuda a probar o desaprobar las afirmaciones que se están probando.
- La prueba de hipótesis también ayuda a crear la estructura del enunciado del problema.
- El proceso proporciona evidencia concluyente para tomar decisiones basadas en el resultado dado por este método.

Hipótesis estadística: cuando un investigador ya tiene una idea sobre el resultado, incluso antes de realizar el experimento. La hipótesis estadística proporciona un marco estructural estándar para trabajar en el problema y tomar una decisión centrada en los datos.

La hipótesis estadística consta de 2 partes:

1. Hipótesis nula $H_{o}$
2. Hipótesis alterna $H_{a}$

La hipótesis nula establece que existe la condición 'Nula', lo que significa que la declaración que se ha dicho está sucediendo, no sucede nada nuevo, las viejas creencias son verdaderas.

La hipótesis alternativa establece que la nueva teoría es verdadera o que algo nuevo está sucediendo, las viejas creencias no son verdaderas.


# Ejemplo

Censo de altura entre estudiantes con una muestra poblacional de 30 personas, la media de la altuna que se obtuvo fue 160 cm. Cual es la regla de desición

$H_{o} = 160$

$H_{a} \ne 160$

En general, si se va a proponer algo nuevo, se define en la hipótesis Alternativa


#### Ejemplo de granta Piscícola (longitud o tamaño de en pulgadas de los peces)

$H_{o} \underline< 15$

$H_{a}  > 15$

El tamaño determina el valor en dolares

$H_{o} = 4$

$H_{a}  > 4$

# Test de probabilidad estadistica


Basado en pocas informaciones basadas en 2 ejemplos, la hipótesis estadística se puede definir en 2 pruebas:
1. Hipótesis de dos colas: las declaraciones no tienen dirección
$$H_{o}: \space \space  \mu = 160$$
$$H_{a}: \space \space  \mu \ne 160$$

este proceso necesita más investigación sobre la dirección de la prueba


2. Pruebas de una cola: Las declaraciones de hipótesis tienen una dirección
$$H_{o}: \space \space  longitud = 15$$
$$H_{a}: \space \space  longitud > 15$$

estas pruebas se usan solo cuando el investigador está seguro de que el resultado de la prueba sería mayor que la información preaprendida


Si se rechaza la Hipótesis Nula y por lo tanto se acepta la Hipótesis Alternativa, se dice que se ha obtenido un resultado estadísticamente significativo, en palabras más sencillas, el resultado obtenido no es pura suerte y se ha tomado la decisión de rechazar la Hipótesis Nula.

En nuestro ejemplo, 2.1 es estadísticamente significativo más alto que 2, pero para los negocios podría no ser significativo como resultado, por lo que se debe tener cuidado al interpretar los resultados de las pruebas estadísticas.

El resultado de la prueba depende de la muestra en consideración, también si un ligero cambio es un resultado sustantivo o no, dependería completamente del caso de uso o del investigador.

# Tipo de error I y II

- Se comete un error tipo l al rechazar una hipótesis nula verdadera.
- Se comete un error de tipo II cuando un investigador empresarial no puede rechazar una hipótesis nula falsa

Tipo de error I  ${\alpha}$

Tipo de error II ${\beta}$

alfa solo puede cometerse cuando se rechaza la hipótesis nula y beta solo puede cometerse cuando no se rechaza la hipótesis nula, un investigador empresarial no puede cometer un error de tipo I y un error de tipo II al mismo tiempo en la misma prueba de hipótesis

beta ocurre solo cuando la hipótesis nula no es verdadera, el cálculo de beta varía
con los muchos parámetros alternativos posibles que podrían ocurrir.

In [ ]:
import pandas as pd
import numpy as np

#llamamos scipy para el proposito estadistico

from scipy import stats
from statsmodels.stats import weightstats as stests

#### Cargaremos los datos

In [ ]:
data = pd.read_excel('/Users/juancamilosierra/Desktop/Clases/Segundo Semestre/Econometria 1/Parcial 1/wage2.xls')

In [ ]:
data.head()

wage  hours   IQ  KWW  educ  exper  tenure  age  married  black  south  \
0   769     40   93   35    12     11       2   31        1      0      0   
1   808     50  119   41    18     11      16   37        1      0      0   
2   825     40  108   46    14     11       9   33        1      0      0   
3   650     40   96   32    12     13       7   32        1      0      0   
4   562     40   74   27    11     14       5   34        1      0      0   

   urban  sibs brthord meduc feduc  
0      1     1       2     8     8  
1      1     1       .    14    14  
2      1     1       2    14    14  
3      1     4       3    12    12  
4      1    10       6     6    11

#### Así que solo a la columna wage usaremos como muestra

In [ ]:
Salario = data['wage']

In [ ]:
Salario

0       769
1       808
2       825
3       650
4       562
       ... 
930     520
931    1202
932     538
933     873
934    1000
Name: wage, Length: 935, dtype: int64

#### Vamos a determinar la media de la muestra (Salario) por lo que tenemos 935 registros, esta es la razon principal para usar la prueba z porque nuestro tamaño es mas de 30 observaciones. En pocas palabras si la muestra tiene menos de 30 observaciones usaremos la prueba t student

In [ ]:
Salario_mean = np.mean(Salario)
print(Salario_mean)

957.9454545454546


In [ ]:
# la media del salario en nuestra muestra es $957.9 dolares

$$z= \frac{\bar{x}-\mu}{\frac{\sigma}{\sqrt n}}$$

$$H_{o}: \space \space  \mu = 957$$
$$H_{a}: \space \space  \mu \ne 957$$

In [ ]:
#Nuestro valor de suposicion sera la regla que determine si rechazo o no rechazo la hipotesis
#es decir nuestra regla de desición  (957)

ztest, pval = stests.ztest(data['wage'], value=957)
print(float(pval))

0.9430035787340623


In [ ]:
if pval<0.05:
    print("Rechazo la hipóptesis nula, es decir hay suficiente evidencia para afirmar Ha")
else:
    print("Acepto la hipotesis nula hay suficiente evidencia afirmar Ho, es decir no hay suficiente evidencia para afirmar Ha ")

Acepto la hipotesis nula hay suficiente evidencia afirmar Ho, es decir no hay suficiente evidencia para afirmar Ha 
